File: src-tauri/src/domain/response/withdraw.rs
use crate::model::withdraw::{
    WithdrawMonthlyAmountResponse, WithdrawResponse, WithdrawResponseDeleteAt,
    WithdrawResponseMonthStatusFailed, WithdrawResponseMonthStatusSuccess,
    WithdrawResponseYearStatusFailed, WithdrawResponseYearStatusSuccess,
    WithdrawYearlyAmountResponse,
};

use super::PaginationMeta;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdrawMonthStatusSuccess {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawResponseMonthStatusSuccess>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdrawYearStatusSuccess {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawResponseYearStatusSuccess>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdrawMonthStatusFailed {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawResponseMonthStatusFailed>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdrawYearStatusFailed {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawResponseYearStatusFailed>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdrawMonthAmount {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawMonthlyAmountResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdrawYearAmount {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawYearlyAmountResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponsesWithdraw {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdraw {
    pub status: String,
    pub message: String,
    pub data: Option<WithdrawResponse>,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdrawDelete {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponseWithdrawAll {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct ApiResponsePaginationWithdraw {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawResponse>,
    pub pagination: PaginationMeta,
}

#[derive(Serialize, Deserialize)]
pub struct ApiResponsePaginationWithdrawDeleteAt {
    pub status: String,
    pub message: String,
    pub data: Vec<WithdrawResponseDeleteAt>,
    pub pagination: PaginationMeta,
}

File: src/types/domain/response/withdraw/withdraw.ts
import {
  Withdraw,
  WithdrawMonthlyAmount,
  WithdrawMonthStatusFailed,
  WithdrawMonthStatusSuccess,
  WithdrawTrashed,
  WithdrawYearlyAmount,
  WithdrawYearStatusFailed,
  WithdrawYearStatusSuccess,
} from "@/types/model";
import { PaginationMeta } from "../pagination";

export interface ApiResponseWithdrawMonthStatusSuccess {
  status: string;
  message: string;
  data: WithdrawMonthStatusSuccess[];
}

export interface ApiResponseWithdrawYearStatusSuccess {
  status: string;
  message: string;
  data: WithdrawYearStatusSuccess[];
}

export interface ApiResponseWithdrawMonthStatusFailed {
  status: string;
  message: string;
  data: WithdrawMonthStatusFailed[];
}

export interface ApiResponseWithdrawYearStatusFailed {
  status: string;
  message: string;
  data: WithdrawYearStatusFailed[];
}

export interface ApiResponseWithdrawMonthAmount {
  status: string;
  message: string;
  data: WithdrawMonthlyAmount[];
}

export interface ApiResponseWithdrawYearAmount {
  status: string;
  message: string;
  data: WithdrawYearlyAmount[];
}

export interface ApiResponsesWithdraw {
  status: string;
  message: string;
  data: Withdraw[];
}

export interface ApiResponseWithdraw {
  status: string;
  message: string;
  data?: Withdraw;
}

export interface ApiResponseWithdrawDelete {
  status: string;
  message: string;
}

export interface ApiResponseWithdrawAll {
  status: string;
  message: string;
}

export interface ApiResponsePaginationWithdraw {
  status: string;
  message: string;
  data: Withdraw[];
  pagination: PaginationMeta;
}

export interface ApiResponsePaginationWithdrawDeleteAt {
  status: string;
  message: string;
  data: WithdrawTrashed[];
  pagination: PaginationMeta;
}
